app/models/purchase.rb

```diff
@@ -289,6 +290,10 @@ def opened_financial_year?
     FinancialYear.on(invoiced_at)&.opened?
   end
 
+  def financial_year_in_closure_preparation?
```

Comment:
Can we find a clearer name? Something like `given_during_closure_preparation?` maybe?